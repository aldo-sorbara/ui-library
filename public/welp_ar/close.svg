<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
