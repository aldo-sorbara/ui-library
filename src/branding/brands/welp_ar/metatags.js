export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
