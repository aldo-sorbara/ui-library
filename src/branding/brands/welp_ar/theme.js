import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
