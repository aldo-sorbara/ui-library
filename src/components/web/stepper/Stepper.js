import React from 'react';
import PropTypes from 'prop-types';
import { Flex, Image } from 'rebass';
import styled, { keyframes, css } from 'styled-components';

const borderfill = keyframes`
    0%   { opacity:0; }
    100% { opacity:1; }
`;

const StepperContainer = styled(Flex)`
  align-items: center;
  background-color: ${props => props.theme.palette.common.bodyBg};
  margin: ${props => props.theme.spacer.stepper.margin};
  padding: 0.5rem 0.5rem;
  position: sticky;
  top: 0;
  z-index: 1021;
`;

const Step = styled(Flex)`
  align-items: center;
  flex-basis: 100%;
  &:last-child {
    flex-basis: 0;
  }
`;

const RoundImage = styled(Image)`
  background-color: ${({ active, theme }) =>
    active ? theme.palette.primary.dark : theme.palette.primary.light};
  border: 2px solid
    ${({ active, theme }) =>
      active ? theme.palette.primary.dark : theme.palette.primary.light};
  border-radius: ${({ size }) => size}px;
  height: 1rem;
  max-width: initial;
  width: 1rem;
`;

RoundImage.defaultProps = {
  theme: {
    palette: {
      primary: {
        dark: '#3a66ab',
        light: '#73baff'
      }
    }
  }
};

const RoundCompleted = styled.div`
  background-color: ${props => props.theme.palette.primary.dark};
  border: 2px solid ${props => props.theme.palette.primary.dark};
  border-radius: 50%;
  flex: none;
  height: 0.5rem;
  transition: border 2s ease-out;
  width: 0.5rem;
`;

RoundCompleted.defaultProps = {
  theme: {
    palette: {
      primary: {
        dark: '#3a66ab'
      }
    }
  }
};

const RoundIcon = styled.div`
  animation: ${borderfill} 0.3s ease-in;
  border-radius: 50%;
  flex: none;
  height: ${({ size }) => size}px;
  ${({ active, theme }) =>
    active
      ? `
        box-shadow: 0px 0px 0px .25rem ${theme.palette.primary.dark} inset;
        width: .75rem;
        height:.75rem;
        `
      : `
        border: 2px solid ${theme.palette.primary.light};
        background-color:#fff;
        width: .5rem;
        height: .5rem

      `}
  transition: box-shadow 0.2s ease-out;
`;

RoundIcon.defaultProps = {
  theme: {
    palette: {
      primary: {
        dark: '#3a66ab',
        light: '#73baff'
      }
    }
  }
};

const lineFillAnim = keyframes`
  0%   { width:0%; }
  50% {width: 50%;}
  100% { width:100%; }
`;

const Line = styled.div`
  background-color: #ccc;
  height: 2px;
  width: 100%;
`;

const DefaultLine = styled(Line)`
  background-color: ${props => props.theme.palette.primary.light};
  margin-left: 2px;
  margin-right: 2px;
  width: 100%;
`;

DefaultLine.defaultProps = {
  theme: {
    palette: {
      primary: {
        light: '#73baff'
      }
    }
  }
};

const lineFillMixin = css`
  animation: ${lineFillAnim} 0.3s ease-out;
`;

const CompleteLine = styled(Line)`
  background-color: ${props => props.theme.palette.primary.dark};
  ${props => props.isCurrentStep && lineFillMixin}
  width: 100%;
`;

CompleteLine.defaultProps = {
  theme: {
    palette: {
      primary: {
        dark: '#3a66ab'
      }
    }
  }
};

const StepRound = ({ active, done, size, last }) => {
  if (done && !last) {
    return <RoundCompleted size={size} />;
  } else if (last) {
    return (
      <RoundImage
        active={active}
        as='img'
        src={
          "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='6.816' height='11.341' viewBox='0 0 6.816 11.341'%3E%3Cdefs%3E%3Cstyle%3E.a%7Bfill:%23fff;%7D%3C/style%3E%3C/defs%3E%3Cpath class='a' d='M7.02-2.532A2.287,2.287,0,0,1,6.354-.876a3.291,3.291,0,0,1-1.794.9V.612a.8.8,0,0,1-.234.6.832.832,0,0,1-.606.228.832.832,0,0,1-.606-.228.8.8,0,0,1-.234-.6V.072A5.391,5.391,0,0,1,.6-.708a.816.816,0,0,1-.306-.3A.941.941,0,0,1,.2-1.452a.972.972,0,0,1,.21-.624.621.621,0,0,1,.5-.264.906.906,0,0,1,.288.048,2.153,2.153,0,0,1,.372.18,4.445,4.445,0,0,0,.984.408,3.807,3.807,0,0,0,.972.12,1.813,1.813,0,0,0,.966-.21.7.7,0,0,0,.33-.63.511.511,0,0,0-.2-.414A1.507,1.507,0,0,0,4.134-3.1q-.294-.1-.858-.24A11.565,11.565,0,0,1,1.8-3.8a2.542,2.542,0,0,1-1-.732,2.008,2.008,0,0,1-.42-1.338,2.343,2.343,0,0,1,.672-1.668,3.292,3.292,0,0,1,1.776-.936v-.6a.794.794,0,0,1,.24-.594.825.825,0,0,1,.6-.234.832.832,0,0,1,.606.228.8.8,0,0,1,.234.6v.564a5.1,5.1,0,0,1,1.944.756.882.882,0,0,1,.306.318.919.919,0,0,1,.09.426.955.955,0,0,1-.216.624.638.638,0,0,1-.5.264.8.8,0,0,1-.276-.048,3.479,3.479,0,0,1-.384-.18q-.084-.048-.372-.2a3.1,3.1,0,0,0-.618-.24,2.658,2.658,0,0,0-.7-.09,1.5,1.5,0,0,0-.876.234.73.73,0,0,0-.336.63.615.615,0,0,0,.384.576,5.877,5.877,0,0,0,1.212.384,10.117,10.117,0,0,1,1.458.432,2.43,2.43,0,0,1,.984.732A2.072,2.072,0,0,1,7.02-2.532Z' transform='translate(-0.204 9.9)'/%3E%3C/svg%3E" // eslint-disable-line
        }
        size={size * 2}
      />
    ); // eslint-disable-line
  }
  return <RoundIcon active={active} size={size} />;
};

const StepLine = ({ done, isCurrentStep }) => (
  <DefaultLine>
    {done && <CompleteLine isCurrentStep={isCurrentStep} />}
  </DefaultLine>
);

const Stepper = props => {
  const { currentStep, size = 8, steps, style } = props;

  const stepsArray = Array(steps).fill();

  return (
    <StepperContainer style={style}>
      {stepsArray.map((el, index) => {
        const params = {
          done: index <= currentStep - 2,
          active: index === currentStep - 1,
          isCurrentStep: index === currentStep - 2,
          size
        };
        return (
          <Step key={`flex-${index}`}>
            <StepRound {...params} last={index === steps - 1} />
            {index !== steps - 1 && <StepLine {...params} />}
          </Step>
        );
      })}
    </StepperContainer>
  );
};

Stepper.propTypes = {
  steps: PropTypes.number.isRequired,
  currentStep: PropTypes.number
};

Stepper.defaultProps = {
  currentStep: 1,
  style: {},
  theme: {
    spacer: {
      stepper: {
        margin: 0
      }
    }
  }
};

export default Stepper;
