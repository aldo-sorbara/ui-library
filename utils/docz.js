export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
