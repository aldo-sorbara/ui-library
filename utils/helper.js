import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
