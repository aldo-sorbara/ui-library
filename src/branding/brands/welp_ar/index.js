import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
