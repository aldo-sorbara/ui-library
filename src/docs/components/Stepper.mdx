---
name: Stepper
menu: Components
route: /stepper
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Stepper from '../../components/web/stepper';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';

# Stepper

<PropsTable of={Stepper} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Stepper steps={4} currentStep={2} />
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/stepper--with-default-props)
