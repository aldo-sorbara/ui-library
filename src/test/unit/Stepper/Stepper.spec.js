import React from 'react';
import Stepper from '../../../components/web/stepper';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Stepper component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Stepper steps={5} />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a default stepper', () => {
    const wrapper = mountWithTheme(<Stepper steps={3} />, welp_ar.theme);
    expect(wrapper.props().steps).toBe(3);
    expect(wrapper.props().currentStep).toBe(1);
  });
  it('should render a Stepper with steps and currentStep', () => {
    const wrapper = mountWithTheme(
      <Stepper steps={5} currentStep={2} />,
      welp_ar.theme
    );
    expect(wrapper.props().steps).toBe(5);
    expect(wrapper.props().currentStep).toBe(2);
  });
  it('should render a complete Stepper', () => {
    const wrapper = mountWithTheme(
      <Stepper steps={5} currentStep={5} />,
      welp_ar.theme
    );
    expect(wrapper.props().steps).toBe(5);
    expect(wrapper.props().currentStep).toBe(5);
  });
});
