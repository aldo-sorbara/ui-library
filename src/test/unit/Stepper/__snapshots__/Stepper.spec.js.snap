// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper component should render snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  padding: 0.5rem 0.5rem;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1021;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c1:last-child {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
}

.c5 {
  box-sizing: border-box;
  max-width: 100%;
  height: auto;
  background-color: #73baff;
  border: 2px solid #73baff;
  border-radius: 16px;
  height: 1rem;
  max-width: initial;
  width: 1rem;
}

.c2 {
  -webkit-animation: bcCCNc 0.3s ease-in;
  animation: bcCCNc 0.3s ease-in;
  border-radius: 50%;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  height: 8px;
  box-shadow: 0px 0px 0px .25rem #3a66ab inset;
  width: .75rem;
  height: .75rem;
  -webkit-transition: box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out;
}

.c4 {
  -webkit-animation: bcCCNc 0.3s ease-in;
  animation: bcCCNc 0.3s ease-in;
  border-radius: 50%;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  height: 8px;
  border: 2px solid #73baff;
  background-color: #fff;
  width: .5rem;
  height: .5rem;
  -webkit-transition: box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out;
}

.c3 {
  background-color: #ccc;
  height: 2px;
  width: 100%;
  background-color: #73baff;
  margin-left: 2px;
  margin-right: 2px;
  width: 100%;
}

<div
  className="c0"
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      size={8}
    />
    <div
      className="c3"
    />
  </div>
  <div
    className="c1"
  >
    <div
      className="c4"
      size={8}
    />
    <div
      className="c3"
    />
  </div>
  <div
    className="c1"
  >
    <div
      className="c4"
      size={8}
    />
    <div
      className="c3"
    />
  </div>
  <div
    className="c1"
  >
    <div
      className="c4"
      size={8}
    />
    <div
      className="c3"
    />
  </div>
  <div
    className="c1"
  >
    <img
      className="c5"
      size={16}
      src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='6.816' height='11.341' viewBox='0 0 6.816 11.341'%3E%3Cdefs%3E%3Cstyle%3E.a%7Bfill:%23fff;%7D%3C/style%3E%3C/defs%3E%3Cpath class='a' d='M7.02-2.532A2.287,2.287,0,0,1,6.354-.876a3.291,3.291,0,0,1-1.794.9V.612a.8.8,0,0,1-.234.6.832.832,0,0,1-.606.228.832.832,0,0,1-.606-.228.8.8,0,0,1-.234-.6V.072A5.391,5.391,0,0,1,.6-.708a.816.816,0,0,1-.306-.3A.941.941,0,0,1,.2-1.452a.972.972,0,0,1,.21-.624.621.621,0,0,1,.5-.264.906.906,0,0,1,.288.048,2.153,2.153,0,0,1,.372.18,4.445,4.445,0,0,0,.984.408,3.807,3.807,0,0,0,.972.12,1.813,1.813,0,0,0,.966-.21.7.7,0,0,0,.33-.63.511.511,0,0,0-.2-.414A1.507,1.507,0,0,0,4.134-3.1q-.294-.1-.858-.24A11.565,11.565,0,0,1,1.8-3.8a2.542,2.542,0,0,1-1-.732,2.008,2.008,0,0,1-.42-1.338,2.343,2.343,0,0,1,.672-1.668,3.292,3.292,0,0,1,1.776-.936v-.6a.794.794,0,0,1,.24-.594.825.825,0,0,1,.6-.234.832.832,0,0,1,.606.228.8.8,0,0,1,.234.6v.564a5.1,5.1,0,0,1,1.944.756.882.882,0,0,1,.306.318.919.919,0,0,1,.09.426.955.955,0,0,1-.216.624.638.638,0,0,1-.5.264.8.8,0,0,1-.276-.048,3.479,3.479,0,0,1-.384-.18q-.084-.048-.372-.2a3.1,3.1,0,0,0-.618-.24,2.658,2.658,0,0,0-.7-.09,1.5,1.5,0,0,0-.876.234.73.73,0,0,0-.336.63.615.615,0,0,0,.384.576,5.877,5.877,0,0,0,1.212.384,10.117,10.117,0,0,1,1.458.432,2.43,2.43,0,0,1,.984.732A2.072,2.072,0,0,1,7.02-2.532Z' transform='translate(-0.204 9.9)'/%3E%3C/svg%3E"
    />
  </div>
</div>
`;
